File: extends/message/domain/message.go
package domain

import "time"

// Message 站内信消息
type Message struct {
	ID          uint   `json:"id"`
	SenderID    uint   `json:"sender_id"`
	SenderType  string `json:"sender_type"` // admin | system（system 预留）
	RecipientID uint   `json:"recipient_id"`
	// RecipientName 收件人用户名（管理端列表关联 users 表填充）
	RecipientName string     `json:"recipient_name,omitempty"`
	Title         string     `json:"title"`
	Content       string     `json:"content"`
	Status        string     `json:"status"` // draft | sent | revoked（draft 预留）
	IsRead        bool       `json:"is_read"`
	ReadAt        *time.Time `json:"read_at"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}

File: java/src/main/java/com/kingfisher/modules/message/domain/Message.java
package com.kingfisher.modules.message.domain;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 站内信消息领域实体，对应 messages 表。
 */
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Message {

    private Long id;
    @JsonProperty("sender_id")
    private Long senderId;
    @JsonProperty("sender_type")
    private String senderType;
    @JsonProperty("recipient_id")
    private Long recipientId;
    @JsonProperty("batch_id")
    private Long batchId;
    @JsonProperty("recipient_name")
    private String recipientName;
    private String title;
    private String content;
    private String status;
    @JsonProperty("is_read")
    private Boolean isRead;
    @JsonProperty("read_at")
    private LocalDateTime readAt;
    @JsonProperty("created_at")
    private LocalDateTime createdAt;
    @JsonProperty("updated_at")
    private LocalDateTime updatedAt;
}
